palantir-java-format/src/main/java/com/palantir/javaformat/java/javadoc/JavadocLexer.java

```diff
@@ -160,13 +162,13 @@ private Type consumeToken() throws LexException {
 
         if (input.tryConsumeRegex(INLINE_TAG_OPEN_PATTERN)) {
             braceDepth.increment();
-            return LITERAL;
+            return INLINE_TAG_OPEN;
         } else if (input.tryConsume("{")) {
             braceDepth.incrementIfPositive();
             return LITERAL;
         } else if (input.tryConsume("}")) {
             braceDepth.decrementIfPositive();
-            return LITERAL;
+            return braceDepth.isPositive() ? LITERAL : INLINE_TAG_CLOSE;
```

Comment:
What happens here with nested inline tags? I noticed there's an `inlineTagDepth` in the other file, but I would have thought that this would only make these tokens for the outermost (from my brief can of these changes)?

Reply:
Do we even care about nested inline tags?

Reply:
we care for situations like
```{@code
class Foo {
}
}<pre>
```

as shown in one of the tests I added, that's actually copied from one of the integration tests where this was initially being handled incorrectly.
Specifically, we want to treat the closing brace of the *class* as a literal (i.e. this is still literal stuff inside the code block), and only the last brace as a INLINE_TAG_CLOSE.

Reply:
On the other hand, in the other fiel I use another one of these depth counters but I don't *think* the number can be anything other than 0 or 1.

Reply:
Makes sense